src/openms/source/FORMAT/MRMFeatureQCFile.cpp

```diff
@@ -28,265 +28,176 @@
-    headers["component_name"] = -1;
-    headers["component_group_name"] = -1;
-    headers["n_heavy_l"] = -1;
-    headers["n_heavy_u"] = -1;
-    headers["n_light_l"] = -1;
-    headers["n_light_u"] = -1;
-    headers["n_detecting_l"] = -1;
-    headers["n_detecting_u"] = -1;
-    headers["n_quantifying_l"] = -1;
-    headers["n_quantifying_u"] = -1;
-    headers["n_identifying_l"] = -1;
-    headers["n_identifying_u"] = -1;
-    headers["n_transitions_l"] = -1;
-    headers["n_transitions_u"] = -1;
-    headers["ion_ratio_pair_name_1"] = -1;
-    headers["ion_ratio_pair_name_2"] = -1;
-    headers["ion_ratio_l"] = -1;
-    headers["ion_ratio_u"] = -1;
-    headers["ion_ratio_feature_name"] = -1;
-    headers["retention_time_l"] = -1;
-    headers["retention_time_u"] = -1;
-    headers["intensity_l"] = -1;
-    headers["intensity_u"] = -1;
-    headers["overall_quality_l"] = -1;
-    headers["overall_quality_u"] = -1;
-    String param_header = "metaValue_";
-    
-    // parse the header columns
-    for (size_t i = 0; i < line.size(); ++i)
-    {
-      // parse transformation_model_params
-      if (line[i].find(param_header) != String::npos) 
-      {
-        line[i].erase(0, param_header.size()); 
-        params_headers[line[i]] = i;
-      }      
-      else // parse all other header entries
+  void MRMFeatureQCFile::pushValuesFromLine_(
+    const StringList& line,
+    const std::map<String, Size>& headers,
+    std::vector<MRMFeatureQC::ComponentQCs>& c_qcs
+  ) const
+  {
+    MRMFeatureQC::ComponentQCs c;
+    std::map<String, Size>::const_iterator it;
+    it = headers.find("component_name");
+    c.component_name = it != headers.end() ? line[it->second] : "";
+    it = headers.find("retention_time_l");
+    c.retention_time_l = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("retention_time_u");
+    c.retention_time_u = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("intensity_l");
+    c.intensity_l = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("intensity_u");
+    c.intensity_u = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("overall_quality_l");
+    c.overall_quality_l = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("overall_quality_u");
+    c.overall_quality_u = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    for (const std::pair<String, Size>& h : headers) // parse the parameters
+    {
+      const String& header = h.first;
+      const Size& i = h.second;
+      boost::smatch m;
+      if (boost::regex_search(header, m, boost::regex("metaValue_(.+)_(l|u)"))) // capture the metavalue name and the boundary and save them to m[1] and m[2]
       {
-        headers[line[i]] = i;
+        setPairValue_(String(m[1]), line[i], String(m[2]), c.meta_value_qc);
       }
     }
+    c_qcs.push_back(c); // TODO: check for any condition before pushing?
```

Comment:
We add defaults for everything so there should not be a need for any condition before adding it to the array.